app/gist/controller.js

```diff
@@ -219,6 +291,28 @@ export default Ember.Controller.extend({
       }
     },
 
+    removeColumn (col) {
+      let numColumns = this.get('realNumColumns');
+
+      for (var i = (col|0); i < numColumns; ++i) {
+        this.setColumnFile(i, this.getColumnFile(i + 1));
+      }
+      this.setColumnFile(numColumns, undefined);
+
+      let activeCol = this.get('activeEditorCol');
+      if (activeCol >= col) {
+        this.set('activeEditorCol', ((activeCol|0) - 1).toString());
+      }
+
+      this.transitionToRoute({queryParams: {numColumns: numColumns - 1}});
+    },
+
+    addColumn() {
+      let numColumns = this.get('realNumColumns');
+
+      this.transitionToRoute({queryParams: {numColumns: numColumns + 1}}).then(this.initializeColumns.bind(this));
```

Comment:
> this.initializeColumns.bind(this)

feels like it should merely be invalidating something, maybe changing columnCount. The explicit `initialColumns` call enforces eager production of something that seems to really want to be pull based.